plugins/grafana-custom-plugins/grafana-dependency-plugin/src/DependencyPanel.tsx

```diff
@@ -47,124 +53,162 @@ export const DependencyPanel: React.FC<Props> = ({ options, data, width, height
+      let dests = new Map<string, number>();
+      dests.set(pod_dst, octetDeltaCount);
+      if (destinationServicePortName !== '') {
+        dests.set(svc_dst, octetDeltaCount);
       }
-      let labels = JSON.parse(labelJSON);
-      if(labels[options.labelName] !== undefined) {
-        return labels[options.labelName];
+      if (srcToDestMap.has(pod_src)) {
+        if (srcToDestMap.get(pod_src)?.has(pod_dst)) {
+          srcToDestMap.get(pod_src)?.set(pod_dst, octetDeltaCount+srcToDestMap.get(pod_src)?.get(pod_dst));
+        } else {
+          srcToDestMap.get(pod_src)?.set(pod_dst, octetDeltaCount);
+        }
+        if (destinationServicePortName === '') {
+          continue;
+        } else if (srcToDestMap.get(pod_src)?.has(svc_dst)) {
+          srcToDestMap.get(pod_src)?.set(svc_dst, octetDeltaCount+srcToDestMap.get(pod_src)?.get(svc_dst));
+        } else {
+          srcToDestMap.get(pod_src)?.set(svc_dst, octetDeltaCount);
+        }
+      } else {
+        srcToDestMap.set(pod_src, dests);
       }
-      return sourcePodName;
     }
+  
+    // format pods inside node within graph string
+    nodeToPodMap.forEach((pods, nodename) => {
+      let str = 'subgraph ' + nodename + '\n';
+      pods.forEach((pod) => {
+        str += nodename + '_pod_' + pod + '(' + pod + ');\n';
+      });
+      str += 'end;\n';
+      graphString += str;
+    });
+    // format arrows to services and pods within graph string
+    let prefixes = ['', 'K', 'M', 'G', 'T'];
+    srcToDestMap.forEach((destsToBytes, src) => {
+      destsToBytes.forEach((bytes, dest) => {
+        let usedpref = Math.floor(Math.log(bytes) / Math.log(1000));
+        if (usedpref > 4) {usedpref = 4};
+        let str = src + ' -- ' + bytes/(Math.pow(1000, usedpref)) + ' ' + prefixes[usedpref] + 'B --> ' + dest + ';\n';
+        graphString += str;
+      });
+    });
+  }
 
-    let groupByPodLabel = options.groupByPodLabel;
-    let srcName = getName(groupByPodLabel, true, sourcePodLabel);
-    let dstName = getName(groupByPodLabel, false, destinationPodLabel);
-
-    // determine which nodes contain which pods
-    if (nodeToPodMap.has(sourceNodeName) && !nodeToPodMap.get(sourceNodeName)?.includes(srcName)) {
-      nodeToPodMap.get(sourceNodeName)?.push(srcName);
-    } else if (!nodeToPodMap.has(sourceNodeName)) {
-      nodeToPodMap.set(sourceNodeName, [srcName]);
-    }
-    if (nodeToPodMap.has(destinationNodeName) && !nodeToPodMap.get(destinationNodeName)?.includes(dstName)) {
-      nodeToPodMap.get(destinationNodeName)?.push(dstName);
-    } else if (!nodeToPodMap.has(destinationNodeName)) {
-      nodeToPodMap.set(destinationNodeName, [dstName]);
+  function layerSevenGraph() {
+    function getColorFromStatus(httpStatus: string) {
+      // colors that correspond to each of the types of http response code; i.e. 4xx and 5xx codes return red to symbolize errors
+      const colors = ['orange', 'green', 'blue', 'red', 'red'];
```

Comment:
I'd suggest just use a map data structure or case switch statement for this function, rather than the status first char matching colors list index.

Reply:
I was thinking of checking the first character because it would allow all response codes that are within the 100 possible codes range to be represented. Since all 4xx codes are error codes, and all 2xx are error codes, I just checked the first character. I can put the underlying logic into a case switch however, that would be cleaner.

Reply:
Checking the first char is ok. But I think the list index thing is not a common practice.

Reply:
I suppose we can combine the first char matching with a case switch statement.